# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Opcodes.h
# Commit: e802face8e04
# Full Hash: e802face8e042d03933c1495d36fd02a9def7dd9
# Author: Brian Hackett <bhackett1024@gmail.com>
# Date: 2019-07-25 21:51:57
# Regressor Bug: 1554524
# File Overlap Count: 1
# Description:
#   Bug 1554524 Part 4 - Add Debugger interface for instrumenting scripts, r=jimb,tcampbell.
# ==============================================================================

diff -r fc204daba6f3 -r e802face8e04 js/src/vm/Opcodes.h
--- a/js/src/vm/Opcodes.h	Mon Jul 08 09:39:52 2019 -1000
+++ b/js/src/vm/Opcodes.h	Mon Jul 08 09:41:21 2019 -1000
@@ -2547,7 +2547,28 @@
      *   Operands: uint32_t constIndex
      *   Stack: => val
      */ \
-    MACRO(JSOP_BIGINT, 237, "bigint", NULL, 5, 0, 1, JOF_BIGINT)
+    MACRO(JSOP_BIGINT, 237, "bigint", NULL, 5, 0, 1, JOF_BIGINT) \
+    /*
+     * Pushes a boolean indicating if instrumentation is active.
+     *   Category: Other
+     *   Operands:
+     *   Stack: => val
+     */ \
+    MACRO(JSOP_INSTRUMENTATION_ACTIVE, 238, "instrumentationActive", NULL, 1, 0, 1, JOF_BYTE) \
+    /*
+     * Pushes the instrumentation callback for the current realm.
+     *   Category: Other
+     *   Operands:
+     *   Stack: => val
+     */ \
+    MACRO(JSOP_INSTRUMENTATION_CALLBACK, 239, "instrumentationCallback", NULL, 1, 0, 1, JOF_BYTE) \
+    /*
+     * Pushes the current script's instrumentation ID.
+     *   Category: Other
+     *   Operands:
+     *   Stack: => val
+     */ \
+    MACRO(JSOP_INSTRUMENTATION_SCRIPT_ID, 240, "instrumentationScriptId", NULL, 1, 0, 1, JOF_BYTE)
 // clang-format on
 
 /*
@@ -2555,9 +2576,6 @@
  * a power of two.  Use this macro to do so.
  */
 #define FOR_EACH_TRAILING_UNUSED_OPCODE(MACRO) \
-  MACRO(238)                                   \
-  MACRO(239)                                   \
-  MACRO(240)                                   \
   MACRO(241)                                   \
   MACRO(242)                                   \
   MACRO(243)                                   \
